generators/client/templates/react/src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx.ejs

```diff
@@ -1,57 +1,175 @@
-                <thead>
-                  <tr>
-                    <th>Blocked Time</th>
-                    <th>Blocked Count</th>
-                    <th>Waited Time</th>
-                    <th>Waited Count</th>
-                    <th>Lock Name</th>
-                  </tr>
-                </thead>
-                <tbody>
-                  <tr key={threadDumpInfo.lockName}>
-                    <td>{threadDumpInfo.blockedTime}</td>
-                    <td>{threadDumpInfo.blockedCount}</td>
-                    <td>{threadDumpInfo.waitedTime}</td>
-                    <td>{threadDumpInfo.waitedCount}</td>
-                    <td>{threadDumpInfo.lockName}</td>
-                  </tr>
-                </tbody>
-              </Table>
-            </div>
-          </div>
+import { Table, Modal, ModalHeader, ModalBody, ModalFooter, Button, Input } from 'reactstrap';
+
+import ThreadItem from './thread-item';
+
+export interface IMetricsModalProps {
+  showModal: boolean;
+  handleClose: Function;
+  threadDump: any;
+}
+
+export interface IMetricsModalState {
+  threadDump: any;
+  displayedThreads: any;
+  threadDumpAll: number;
+  threadDumpBlocked: number;
+  threadDumpRunnable: number;
+  threadDumpTimedWaiting: number;
+  threadDumpWaiting: number;
+}
+
+export class MetricsModal extends React.Component<IMetricsModalProps, IMetricsModalState> {
+  state: IMetricsModalState = {
+    threadDump: {},
+    displayedThreads: [],
+    threadDumpAll: 0,
+    threadDumpBlocked: 0,
+    threadDumpRunnable: 0,
+    threadDumpTimedWaiting: 0,
+    threadDumpWaiting: 0
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.threadDump.threads && nextProps.threadDump !== this.props.threadDump) {
+      this.setState({
+        threadDump: nextProps.threadDump,
+        displayedThreads: nextProps.threadDump.threads
+      }, () => {
+        this.refreshThreadsCounter();
+      });
+    }
+  }
+
+  refreshThreadsCounter = () => {
+    let threadDumpAll = 0;
+    let threadDumpRunnable = 0;
+    let threadDumpWaiting = 0;
+    let threadDumpTimedWaiting = 0;
+    let threadDumpBlocked = 0;
+
+    this.state.threadDump.threads.forEach(t => {
+      switch (t.threadState) {
+        case 'RUNNABLE':
+          threadDumpRunnable++;
+          break;
+        case 'WAITING':
+          threadDumpWaiting++;
+          break;
+        case 'TIMED_WAITING':
+          threadDumpTimedWaiting++;
+          break;
+        case 'BLOCKED':
+          threadDumpBlocked++;
+          break;
+        default:
+          break;
+      }
+    });
+
+    threadDumpAll = threadDumpRunnable + threadDumpWaiting + threadDumpTimedWaiting + threadDumpBlocked;
+    this.setState({ threadDumpAll, threadDumpRunnable, threadDumpWaiting, threadDumpTimedWaiting, threadDumpBlocked });
+  };
+
+  searchFilter = event => {
+    let filteredList = this.state.threadDump.threads;
+    filteredList = filteredList.filter(t => t.lockName != null &&
```

Comment:
set the search value in state and do the actual filtering during rendering and avoid the `displayedThreads` state if possible